rectools/models/nn/bert4rec.py

```diff
@@ -82,7 +82,13 @@ def _collate_fn_train(
         self,
         batch: List[Tuple[List[int], List[float]]],
     ) -> Dict[str, torch.Tensor]:
-        """TODO"""
+        """
+        Mask session elements to receive `x`.
+        Get target by replacing session elements with 0 with probability 1 - `mask_prob`.
+        Truncate each session and target from right to keep (`session_max_len` + 1) last items.
+        Do left padding until  (session_max_len + 1) is reached.
+        If `n_negatives` is not None, generate negative items from uniform distribution.
```

Comment:
`session_max_len + 1` is used here to keep that logic that `session_max_len` is maximum sessions length that model will process during inference. But one extra "MASK" token will be added for making predictions. Thus we need to see `session_max_len + 1` elements during training.